### PR TITLE
Add condition for sidecar unreachable

### PR DESCRIPTION
**Pull request by johscheuer**
# Description

Fixes: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/716

## Type of change

*Please select one of the options below.*

- New feature (non-breaking change which adds functionality)

# Discussion

*Are there any design details that you would like to discuss further?*

# Testing

Locally and e2e.

*Do we need to perform additional testing once this is merged, or perform in a larger testing environment?*

# Documentation

None

# Follow-up

None.

### CODE REVIEW

File: controllers/cluster_controller.go [hunk 1]
```diff
@@ -166,10 +167,18 @@ func (r *FoundationDBClusterReconciler) Reconcile(ctx context.Context, request c
 			continue
 		}
 
+		if requeue.DelayedRequeue {
```
Comment (johscheuer) [inline controllers/cluster_controller.go:170]: I don't make use of that yet, but the plan is to use this for delayed requeue e.g. when we want to iterate further but in the end queue the cluster again.

File: controllers/controllers.go [hunk 2]
```diff
@@ -109,9 +113,10 @@ func processRequeue(requeue *Requeue, subReconciler interface{}, object runtime.
 	recorder.Event(object, corev1.EventTypeNormal, "ReconciliationTerminatedEarly", requeue.Message)
 
 	if err != nil {
-		logger.Error(err, "Error in reconciliation", "subReconciler", fmt.Sprintf("%T", subReconciler), "requeueAfter", requeue.Delay)
+		curLog.Error(err, "Error in reconciliation")
 		return ctrl.Result{}, err
 	}
-	logger.Info("Reconciliation terminated early", "subReconciler", fmt.Sprintf("%T", subReconciler), "message", requeue.Message, "requeueAfter", requeue.Delay)
+	logger.Info("Reconciliation terminated early", "message", requeue.Message)
```
Comment (johscheuer) [inline controllers/controllers.go:119]:
```suggestion
	curLog.Info("Reconciliation terminated early", "message", requeue.Message)
```

File: controllers/update_pod_config.go [hunk 3]
```diff
@@ -65,13 +86,22 @@ func (u UpdatePodConfig) Reconcile(r *FoundationDBClusterReconciler, context ctx
 		synced, err := r.updatePodDynamicConf(cluster, instance)
 		if !synced {
 			allSynced = false
-			log.Info("Update dynamic Pod config", "namespace", cluster.Namespace, "cluster", cluster.Name, "processGroupID", instance.GetInstanceID(), "synced", synced, "error", err)
+			hasUpdate = true
+			curLogger.Info("Update dynamic Pod config", "synced", synced, "error", err)
+
+			if err != nil {
+				if _, ok := err.(net.Error); ok {
```
Comment (johscheuer) [inline controllers/update_pod_config.go:93]: I think we should iterate over the error and unwrap until we hit the end.

File: controllers/update_status_test.go [hunk 4]
```diff
@@ -380,5 +380,53 @@ var _ = Describe("update_status", func() {
 				Expect(removalCount).To(BeNumerically("==", 1))
 			})
 		})
+
+		When("one instance is not reachable", func() {
+			var unreachableProcessGroup string
+
+			BeforeEach(func() {
+				unreachableProcessGroup = instances[0].GetInstanceID()
+				instances[0].Pod.Annotations[mockUnreachableAnnotation] = "banana"
```
Comment (09harsh) [inline controllers/update_status_test.go:389]: Shouldn't we need to clear this after test completes

Comment (johscheuer) [reply controllers/update_status_test.go:389]: We create a complete new cluster object for each test so we don't have to remove anything.

**Pull request merged**